src/main/scripts/jbang.cmd

```diff
@@ -20,20 +20,45 @@ set OUTPUT=
 
 rem create TDIR based on jbang_dir in case it is missing to have a folder
 rem we need to be able to write to anyway
-IF "%JBANG_DIR%"=="" (set TDIR=%userprofile%\.jbang) ELSE (set TDIR=%JBANG_DIR%\.jbang)
+if "%JBANG_DIR%"=="" (set JBDIR=%userprofile%\.jbang) else (set JBDIR=%JBANG_DIR%)
+if "%JBANG_CACHE_DIR%"=="" (set TDIR=%JBDIR%\cache) else (set TDIR=%JBANG_CACHE_DIR%)
 
-IF NOT EXIST "%TDIR%" ( mkdir "%TDIR%")
+if not exist "%TDIR%\jdks" ( mkdir "%TDIR%\jdks" )
+
+rem prefer JAVA instead of PATH to resolve `java` location
+rem if [[ -z "$JAVA_HOME" ]]; then JAVA_EXEC="java"; else JAVA_EXEC="$JAVA_HOME/bin/java"; fi
+if "%JAVA_HOME%"=="" (
+  where javac 2>&1 > nul
+  if %errorlevel% equ 0 (
+    set JAVA_EXEC=java.exe
+  ) else (
+    if not exist "%TDIR%\jdks\%javaVersion%" (
+      echo Downloading JDK %javaVersion%...
+      powershell -NonInteractive -Command "$ProgressPreference = 'SilentlyContinue'; Invoke-WebRequest %url% -OutFile %TDIR%\bootstrap-jdk.zip"
+      echo Installing JDK %javaVersion%...
+      if exist "%TDIR%\jdks\%javaVersion%.tmp" ( rd /s /q "%TDIR%\jdks\%javaVersion%.tmp" 2>&1 > nul )
+      powershell -NonInteractive -Command "$ProgressPreference = 'SilentlyContinue'; Expand-Archive -Path %TDIR%\bootstrap-jdk.zip -DestinationPath %TDIR%\jdks\%javaVersion%.tmp"
+	  for /d %%d in (%TDIR%\jdks\%javaVersion%.tmp\*) do (
+        powershell -NonInteractive -Command "Move-Item %%d\* !TDIR!\jdks\%javaVersion%.tmp"
+	  )
+      ren "%TDIR%\jdks\%javaVersion%.tmp" "%javaVersion%"
+    )
+    set JAVA_HOME=%TDIR%\jdks\%javaVersion%
+    set JAVA_EXEC=!JAVA_HOME!\bin\java.exe
+  )
+) else (
+  set JAVA_EXEC="%JAVA_HOME%\bin\java.exe"
+)
```

Comment:
testing on windows i get: "INFO: Could not find files for the given pattern(s)." somewhere in this block (windows won't tell which)

Reply:
This should work now.

Reply:
it seems to do ;)